framework/caching/CacheInterface.php

```diff
@@ -0,0 +1,224 @@
+<?php
+
+namespace yii\caching;
+
+/**
+ * CacheInterface is the base interface for cache.
+ *
+ * A data item can be stored in the cache by calling [[set()]] and be retrieved back
+ * later (in the same or different request) by [[get()]]. In both operations,
+ * a key identifying the data item is required. An expiration time and/or a [[Dependency|dependency]]
+ * can also be specified when calling [[set()]]. If the data item expires or the dependency
+ * changes at the time of calling [[get()]], the cache will return no data.
+ *
+ * A typical usage pattern of cache is like the following:
+ *
+ * ```php
+ * $key = 'demo';
+ * $data = $cache->get($key);
+ * if ($data === false) {
+ *     // ...generate $data here...
+ *     $cache->set($key, $data, $duration, $dependency);
+ * }
+ * ```
+ *
+ * Because CacheInterface extends the [[\ArrayAccess]] interface, it can be used like an array. For example,
+ *
+ * ```php
+ * $cache['foo'] = 'some data';
+ * echo $cache['foo'];
+ * ```
+ *
+ * For more details and usage information on Cache, see the [guide article on caching](guide:caching-overview).
+ *
+ * @author Qiang Xue <qiang.xue@gmail.com>
+ * @author Dmitry Naumenko <d.naumenko.a@gmail.com>
+ * @since 2.0.13. Previous framework versions used abstract class [[yii\caching\Cache]] as interface.
+ */
+interface CacheInterface extends \ArrayAccess
+{
+    /**
+     * Builds a normalized cache key from a given key.
+     *
+     * If the given key is a string containing alphanumeric characters only and no more than 32 characters,
+     * then the key will be returned back prefixed with [[keyPrefix]]. Otherwise, a normalized key
+     * is generated by serializing the given key, applying MD5 hashing, and prefixing with [[keyPrefix]].
+     *
+     * @param mixed $key the key to be normalized
+     * @return string the generated cache key
+     */
+    public function buildKey($key);
```

Comment:
It was protected, but in July 2014 Qiang turned it to public for some reasons 2e71f6e

I do not think this method is vital, but I've added it to the interface because interface API should be compatible with current abstract class Cache. Don't you agree?